File: examples/node/ingest-sbom.ts
import { readFile } from "node:fs/promises";

const apiBaseUrl = process.env.API_BASE_URL;
const serviceApiKey = process.env.SERVICE_API_KEY;
const productId = process.env.PRODUCT_ID;
const productVersion = process.env.PRODUCT_VERSION ?? "1.0.0";
const sbomFile = process.env.SBOM_FILE ?? "./sbom.json";

if (!apiBaseUrl || !serviceApiKey || !productId) {
  throw new Error("Set API_BASE_URL, SERVICE_API_KEY, and PRODUCT_ID");
}

const form = new FormData();
form.set("version", productVersion);
form.set("files", new Blob([await readFile(sbomFile)], { type: "application/json" }), "sbom.json");

const response = await fetch(`${apiBaseUrl}/ingest-sbom?queue_analysis=true`, {
  method: "POST",
  headers: {
    Authorization: `Bearer ${serviceApiKey}`,
    "X-Product-Id": productId,
  },
  body: form,
});

console.log(JSON.stringify(await response.json(), null, 2));

File: examples/python/ingest_sbom.py
import os
import subprocess

api_base_url = os.environ["API_BASE_URL"]
service_api_key = os.environ["SERVICE_API_KEY"]
product_id = os.environ["PRODUCT_ID"]
product_version = os.environ.get("PRODUCT_VERSION", "1.0.0")
sbom_file = os.environ.get("SBOM_FILE", "./sbom.json")

subprocess.run(
    [
        "curl",
        "-sS",
        "-X",
        "POST",
        f"{api_base_url}/ingest-sbom?queue_analysis=true",
        "-H",
        f"Authorization: Bearer {service_api_key}",
        "-H",
        f"X-Product-Id: {product_id}",
        "-F",
        f"version={product_version}",
        "-F",
        f"files=@{sbom_file};type=application/json",
    ],
    check=True,
)

